### ISSUE TITLE
Model.Url(culture) only returns "#" on multi-lingual site

### ISSUE THREAD

**Issue by jattwood** [umbraco/Umbraco-CMS#11687]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

9.1.0

### Bug summary

Using Umbraco 9.1.0, when trying to get the URL for a page in the context of the current culture, Model.Url(culture) will only return "#" instead of the actual page URL. Removing the culture and using the base method Model.Url() will correctly return the default culture URL.

### Specifics

Using the following code to build a language picker on a multi-lingual site:

```
<div>
    Choose language:<br>
    @foreach (var (culture, infos) in Model.Cultures)
    {
        <a href="@Model.Url(culture)">
            @culture.ToFriendlyName()
        </a>
    }
</div>
```

### Steps to reproduce

1) Set up a basic 2 language multi-lingual site in my case the cultures are en-us and fr-ca for English and Canadian French.
2) Enable Models Builder ( I am using SourceCodeManual in a separate class library if that matters)
3) Create a very basic document type for a home page (and one sub page if you wish) that has "Allow vary by culture" set a the document type level and the property level.
4) In culture and host names on the home node, set /en and /fr for each language culture respectively.
5) Save and publish both EN and FR content for the created doc types.
6) in the respective template, use the following code to output cultures and their respective URLs:

```
<div>
    Cultures Loop:<br>
    @foreach (var (culture, infos) in Model.Cultures)
    {
<ul>
    <li>Culture: @culture.ToString()</li>
    <li>Model.Url(culture): @Model.Url(culture)</li>
    <li>Model.Url(culture): @Model.UrlSegment(culture)</li>
</ul>
    }
</div>
```
7) when rendering the page, you should see the correct cultures getting outputted, however, their URLs will be "#":
![image](https://user-images.githubusercontent.com/6452616/143061517-4ffd2e53-4092-4cbd-b739-4e9321bc5c17.png)

### Expected result / actual result

Expected result would be that the correct URL for each culture would be outputted by the URL method. Note that UrlSegment() does return the correct result.

---
_This item has been added to our backlog [AB#15958](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/15958)_

**Comment by ronaldbarendse** [comment]
This might happen when the content itself or any of its parents isn't published in that culture: can you check what the following outputs: `Model.IsPublished(culture)`?

**Comment by jattwood** [comment]
Thanks @ronaldbarendse for the suggestion. I have tried that now as well - here's the output below.
It's puzzling because, @Model.Url() with no culture returns the correct page URL. It's just when you feed it a culture that it gaps out...

![image](https://user-images.githubusercontent.com/6452616/143454104-a6b43d4f-857a-4629-b326-558e4d4371cd.png)

**Comment by thiago2803** [comment]
FYI
This is a regression from the 9.0 to 9.1.0

https://our.umbraco.com/forum/umbraco-9/107682-different-output-result-from-90-to-91-when-using-url-culture

For some reason the culture in 9.1 is case sensitve

**Comment by jattwood** [comment]
Thanks for that info @thiago2803.

I should think that we should be able to pass culture into url() as an object without having to worry about casing.

I have delved pretty deep into the source and could not see where that's happening looks like it goes as far down into nu-cache. I'll look again with that lense. 

Cheers

Jamie

**Comment by thiago2803** [comment]
Hi Jamie

I agree with you that we should not worry about casing

The issue is here: 

https://github.com/umbraco/Umbraco-CMS/blob/v9/contrib/src/Umbraco.Core/Routing/DefaultUrlProvider.cs#L157 

`culture == defaultCulture`

**Comment by jattwood** [comment]
Thank you! I will try a PR tomorrow. I have yet to finish setting up v9 from source to test this scenario but will try to see what can be done ASAP. Cheers!

**Comment by ronaldbarendse** [comment]
Good find @thiago2803 👍🏻 The content cache (NuCache) should already be case-insensitive and I see the `IsPublished(culture)` is already fixed in https://github.com/umbraco/Umbraco-CMS/pull/10959, which is why the result for both is already `True`.

**Comment by jattwood** [comment]
The casing is only part of the issue. Even with something like this:

`var defaultCulture = _localizationService.GetDefaultLanguageIsoCode().ToLower();`

This only will fix part of the equation and will only pass line 157 if the passed culture == the default culture (in my case if en-us == en-us).  

If we are on the home page or site root node using the culture and host names "/en" and "/fr" and the passed culture does not equal the default culture (fr-ca == en-us) then the inner contents of the if statement are never reached because var pos = 0 (null), culture == defaultCulture = false and culture is null = false.

I have gone back through all the commits and can't see where this was introduced. This must have existed back into 8.x...?

**Comment by nikolajlauridsen** [comment]
Thanks for reporting 🐛, this has been fixed in #11886
